kakarot_scripts/utils/starknet.py

```diff
@@ -387,18 +400,43 @@ async def deploy_starknet_account(class_hash=None, private_key=None, amount=1):
         constructor_calldata=constructor_calldata,
         deployer_address=0,
     )
-    logger.info(f"ℹ️  Funding account {hex(address)} with {amount} ETH")
-    await fund_address(address, amount=amount)
-    logger.info("ℹ️  Deploying account")
-    res = await Account.deploy_account_v1(
-        address=address,
-        class_hash=class_hash,
-        salt=salt,
-        key_pair=key_pair,
-        client=RPC_CLIENT,
-        constructor_calldata=constructor_calldata,
-        max_fee=_max_fee,
-    )
+    current_balance = await get_balance(address)
+    required_amount = amount - current_balance
+    if required_amount > 0:
+        await fund_address(address, amount=required_amount)
+
+    try:
+        await RPC_CLIENT.get_class_hash_at(address)
+        logger.info(f"✅ Account at 0x{address:064x}")
+        return {
+            "address": address,
+        }
+    except Exception:
+        pass
+
+    logger.info(f"ℹ️  Deploying account with salt {hex(salt)}")
+    try:
+        res = await Account.deploy_account_v1(
+            address=address,
+            class_hash=class_hash,
+            salt=salt,
+            key_pair=key_pair,
+            client=RPC_CLIENT,
+            constructor_calldata=constructor_calldata,
+            max_fee=_max_fee,
+        )
+    except Exception as e:
+        logger.info(f"❌ Error deploying account: {e}, waiting 5s before retrying")
```

Comment:
dumb question but why don't we logger.warn when something wrong happens in general?